src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -5383,8 +5383,14 @@ NOINLINE static void JIT_ReversePInvokeEnterRare(ReversePInvokeFrame* frame)
     if (thread->PreemptiveGCDisabled())
         ReversePInvokeBadTransition();
 
-    thread->DisablePreemptiveGC();
     frame->currentThread = thread;
```

Comment:
>frame->currentThread = thread; [](start = 4, length = 30)

Did this need to move before the call to `thread->DisablePreemptiveGC();`?

Reply:
`frame->currentThread` is set before calling `DisablePreemptiveGC` in the other path. I have done this to make both paths symetric.